typing/ctype.ml

```diff
@@ -1750,6 +1750,13 @@ let full_expand ~may_forget_scope env ty =
   | _ ->
       ty
 
+let rec full_expand_fun env ty =
+  match get_desc ty with
+  | Tarrow (lbl, lhs, rhs, commutable) ->
+      let rhs = full_expand_fun env (expand_head env rhs) in
+      newty (Tarrow (lbl, lhs, rhs, commutable))
+  | _ -> ty
```

Comment:
The function is only correct because we are in a context where we know that we don't have a infinite arrow (int -> 'a as 'a`), and we will only use the type for printing. Thus I propose to move back this function to `typecore` as an auxiliary function for error reporting.

Reply:
That wasn't ready to be in the PR.